Bottom: 29ecc3bc290c024c759302d9dc169615b26d2053
Top:    798fc727e6326a3f9274398668c07e1ad23db8c5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:22:14 +0000

BORAST: Fixup to only draw the passed POLYAREA, not walk the linked list of POLYAREA and draw them all!


---

diff --git a/src/borast/borast-bentley-ottmann.c b/src/borast/borast-bentley-ottmann.c
index aedbb37..4122b66 100644
--- a/src/borast/borast-bentley-ottmann.c
+++ b/src/borast/borast-bentley-ottmann.c
@@ -1383,6 +1383,7 @@ poly_area_to_start_events (POLYAREA                *poly,
                                  counter, outer_contour);
         outer_contour = 0;
       }
+      break;
 
     } while ((pa = pa->f) != poly);
 
@@ -1409,6 +1410,7 @@ bo_poly_to_traps (hidGC gc, POLYAREA *poly, borast_traps_t *traps)
     for (contour = pa->contours; contour != NULL; contour = contour->next)
       num_events += contour->Count;
     /* FIXME: Remove horizontal edges? */
+    break;
   } while ((pa = pa->f) != poly);
 
   if (unlikely (0 == num_events))
